libavcodec/libaribcaption.c

```diff
@@ -0,0 +1,1141 @@
+            return AVERROR_EXTERNAL;
+        }
+        if(!aribcc_renderer_initialize(ctx->renderer,
+                                       ARIBCC_CAPTIONTYPE_CAPTION,
+                                       ARIBCC_FONTPROVIDER_TYPE_AUTO,
+                                       ARIBCC_TEXTRENDERER_TYPE_AUTO)) {
+            av_log(avctx, AV_LOG_ERROR, "Failed to initialize libaribcaption renderer.\n");
+            aribcaption_close(avctx);
+            return AVERROR_EXTERNAL;
+        }
+        estimate_video_frame_size(ctx);
+        if (!aribcc_renderer_set_frame_size(ctx->renderer,
+                                            ctx->frame_width, ctx->frame_height)) {
+            av_log(ctx, AV_LOG_ERROR,
+                   "aribcc_renderer_set_frame_size() returned with error.\n");
+            aribcaption_close(avctx);
+            return AVERROR_EXTERNAL;
+        }
+        if (!(ctx->clut = av_mallocz(AVPALETTE_SIZE))) {
+            aribcaption_close(avctx);
+            return AVERROR(ENOMEM);
+        }
+
+        aribcc_renderer_set_storage_policy(ctx->renderer, ARIBCC_CAPTION_STORAGE_POLICY_MINIMUM, 0);
+        aribcc_renderer_set_replace_drcs(ctx->renderer, ctx->replace_drcs);
+        aribcc_renderer_set_force_stroke_text(ctx->renderer, ctx->force_stroke_text);
+        aribcc_renderer_set_force_no_background(ctx->renderer, ctx->ignore_background);
+        aribcc_renderer_set_force_no_ruby(ctx->renderer, ctx->ignore_ruby);
+        aribcc_renderer_set_stroke_width(ctx->renderer, ctx->stroke_width);
+        if (ctx->font) {
+            int is_nomem = 0;
+            size_t count = 0;
+            const char **font_families = NULL;
+            const char *fonts = ctx->font;
+
+            while (!is_nomem && *fonts) {
+                const char **ff = av_realloc_array(font_families, count + 1, sizeof(*font_families));
+                if (!ff)
+                    is_nomem = 1;
+                else {
+                    font_families = ff;
+                    ff[count++] = av_get_token(&fonts, ",");
+                    if (!ff[count - 1])
+                        is_nomem = 1;
+                    else if (*fonts)
+                        fonts++;
+                }
+            }
+            if (!is_nomem && count)
+                aribcc_renderer_set_default_font_family(ctx->renderer, font_families, count, true);
+            while (count)
+                av_freep(&font_families[--count]);
+            av_freep(&font_families);
+            if (is_nomem) {
+                aribcaption_close(avctx);
+                return AVERROR(ENOMEM);
+            }
+        }
+        break;
+
+    case SUBTITLE_TEXT:
+    case SUBTITLE_NONE:
+    default:
+        /* do nothing */ ;
+    }
+
+    ctx->readorder = 0;
+
+    return 0;
+}
+
+#if !defined(DEFAULT_SUBTITLE_TYPE)
+# define DEFAULT_SUBTITLE_TYPE SUBTITLE_ASS
+#endif
+#if !defined(ASS_WORKAROUND)
+# define ASS_WORKAROUND 1
+#endif
```

Comment:
These should simply be set directly in the `AVOption` array.

Reply:
These provide only default behavior.
Former one can be embeded in the `AVOption` array. The default behavior would not be changed.
Latter one depends when most of software (or major software) can handle multi-rectangle ASS subtitle. Thus, I prefer to put it out from the `AVOption` array.